From 96f0e7d8e24a4eb72210a967977c8b5d3ced21b1 Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Fri, 17 Aug 2012 10:16:42 +0300
Subject: [PATCH 291/311] conf/reg.c: pass a pointer, not a struct itself

c2_conf_reg_lookup() and c2_conf_buf_eq() expect a *pointer* to c2_conf_buf.

LogD #752
---
 conf/reg.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/conf/reg.c b/conf/reg.c
index 9e777e7..b542092 100644
--- a/conf/reg.c
+++ b/conf/reg.c
@@ -85,7 +85,7 @@ void c2_conf_reg_fini(struct c2_conf_reg *reg)
 int c2_conf_reg_add(struct c2_conf_reg *reg, struct c2_conf_obj *obj)
 {
 	struct objptr *p;
-	C2_PRE(c2_conf_reg_lookup(reg, obj->co_type, obj->co_id) == NULL);
+	C2_PRE(c2_conf_reg_lookup(reg, obj->co_type, &obj->co_id) == NULL);
 
 	C2_ALLOC_PTR(p);
 	if (p == NULL)
@@ -106,7 +106,7 @@ struct c2_conf_obj *c2_conf_reg_lookup(const struct c2_conf_reg *reg,
 
 	c2_tlist_for(&reg_tl, &reg->r_ptrs, p) {
 		if (p->p_obj->co_type == type &&
-		    c2_conf_buf_eq(p->p_obj->co_id, id))
+		    c2_conf_buf_eq(&p->p_obj->co_id, id))
 			return p->p_obj;
 	} c2_tlist_endfor;
 
-- 
1.8.3.2

